### PR TITLE
Fix tests for DumperTest

### PR DESCRIPTION
**Pull request by Toflar**
Collation and engine defaults have been removed, they will be derived from the connection apparently: https://github.com/doctrine/dbal/commit/86df5dfdbad75b9726cfee7e712aa0c3b6469fec

### CODE REVIEW

**Comment by Toflar** [comment]
Lots of this tests have been written by @ausi so he should review this for sure.

File: core-bundle/tests/Doctrine/Backup/DumperTest.php [hunk 1]
```diff
@@ -264,7 +264,7 @@ public function successfulDumpProvider(): \Generator
                 'SET FOREIGN_KEY_CHECKS = 0;',
                 '-- BEGIN STRUCTURE tl_page',
                 'DROP TABLE IF EXISTS `tl_page`;',
-                "CREATE TABLE `tl_page` (`x_string` VARCHAR(255) CHARACTER SET utf8mb4 NOT NULL, `x_array` LONGTEXT NOT NULL COMMENT '(DC2Type:array)', `x_ascii` VARCHAR(255) NOT NULL, `x_binary` VARBINARY(255) NOT NULL, `x_blob` LONGBLOB NOT NULL, `x_object` LONGTEXT NOT NULL COMMENT '(DC2Type:object)', `x_simple_array` LONGTEXT NOT NULL COMMENT '(DC2Type:simple_array)') DEFAULT CHARACTER SET utf8 COLLATE `utf8_unicode_ci` ENGINE = InnoDB;",
```
Comment (Toflar) [inline core-bundle/tests/Doctrine/Backup/DumperTest.php:267]: @ausi the deprecated docs say one should use `Types::JSON` for what used to be `Types::OBJECT` and `Types::ARRAY`. Not sure those changes are correct. `TYPES::ARRAY` to JSON sounds legit. But `Types::OBJECT` - not sure.

Comment (ausi) [reply core-bundle/tests/Doctrine/Backup/DumperTest.php:267]: Should be fine I think.
I’d probably rename `x_array` to `x_json` and remove the `x_object` column then.

File: core-bundle/tests/Doctrine/Backup/DumperTest.php [hunk 2]
```diff
@@ -68,7 +68,7 @@ public function testUnsuccessfulDump(): void
     public function successfulDumpProvider(): \Generator
     {
         yield 'Empty table without data' => [
-            [new Table('tl_page', [new Column('foobar', Type::getType(Types::STRING))])],
+            [new Table('tl_page', [new Column('foobar', Type::getType(Types::STRING), ['length' => 255])])],
```
Comment (ausi) [inline core-bundle/tests/Doctrine/Backup/DumperTest.php]:
```suggestion
            [new Table('tl_page', [new Column('foobar', Type::getType(Types::STRING), ['length' => 255])], [], [], [], ['charset' => 'utf8', 'collation' => 'utf8_unicode_ci', 'engine' => 'InnoDB'])],
```

I think we should test if the options still get dumpted.

**Comment by ausi** [comment]
We should also make sure that we have a functional test that ensures that we still dump `charset`, `collation` and `engine`.

**Comment by Toflar** [comment]
All done.

**Pull request merged**